python/ml4ir/base/tests/test_auto_dag_network.py

```diff
@@ -0,0 +1,117 @@
+import unittest
+
+from tensorflow.keras.layers import Layer
+
+from ml4ir.base.model.architectures.auto_dag_network import LayerNode, get_layer_subclasses, CycleFoundException, \
+    LayerGraph
+
+
+class UserDefinedTestLayerGlobal(Layer):
+    pass
+
+
+class TestGetLayerSubclasses(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.layer_subclasses = get_layer_subclasses()
+
+    def test_local_userdefined_subclass(self):
+        """Test to show local classes are also accessible but need to be in scope to be found"""
+
+        class UserDefinedTestLayerLocal(Layer):
+            pass
+
+        self.assertNotIn(UserDefinedTestLayerLocal, self.layer_subclasses.values())
+        self.assertIn(UserDefinedTestLayerLocal, get_layer_subclasses().values())
```

Comment:
@arvindsrikantan Worth adding a test for the string to be used in model config to access this.